internal/parser/translate.go

```diff
@@ -327,84 +327,155 @@ func fromIngressV1(log logrus.FieldLogger, ingressList []*networkingv1.Ingress)
 
 // fromHTTPRoutes processes all the HTTPRoute objects present in the cache and translates
 // them to Kong Gateway configurations.
-//
-// TODO: this function is a WIP and is only implemented at POC level currently. The code
-//       is not reachable unless the Gateway feature gate has been enabled and the relevant
-//       milestone for tracking completion is:
-//
-//         https://github.com/Kong/kubernetes-ingress-controller/issues/2079
-//
-//       you may see several TODO's scattered throughout this function, they are all covered by #2079
-func fromHTTPRoutes(_ logrus.FieldLogger, httpRouteList []*gatewayv1alpha2.HTTPRoute) ingressRules { // TODO: use log
+func fromHTTPRoutes(log logrus.FieldLogger, httpRouteList []*gatewayv1alpha2.HTTPRoute) ingressRules {
```

Comment:
I think we should unit test this functions like we do other `from*` functions in this file.

These unit tests seem like the right place for testing all logic paths through the translate logic, and we don't seem to be there.

Reply:
As per the description, I intended for follow up PRs to include more tests. Given that I expect to significantly refactor this code once integration tests are complete I would like to wait on the unit testing until I've done the code refactor. I've added the bullet point `- [ ] unit tests for refactored parser functions` to https://github.com/Kong/kubernetes-ingress-controller/issues/2079, let me know if you feel that will be sufficient?

Reply:
as discussed offline - I do stand by the rule that unit testing belongs closer to code than this, but for sake of not disrupting you until we discuss this team-wide, I'm fine with making this an AC on #2079 for now.